assign-profile.lua

```diff
@@ -155,63 +146,40 @@ end
 
 -- ------------------------------------------------------ MAIN ------------------------------------------------------ --
 local function main(...)
-    local args = { ... }
+    local args = utils.processArgs({...}, valid_args)
 
-    if #args == 0 then
+    if args.help then
         print(help)
         return
     end
 
-    local unit_id
-    local filename
-    local profile_name
-    local reset_table = {}
-
-
-    local i = 1
-    while i <= #args do
-        local arg = args[i]
-        if arg == valid_args.HELP then
-            print(help)
-            return
-        elseif arg == valid_args.UNIT then
-            i = i + 1 -- consume next arg
-            local unit_id_str = args[i]
-            if not unit_id_str then
-                -- we reached the end of the arguments list
-                qerror("Missing unit id.")
-            end
-            unit_id = tonumber(unit_id_str)
-            if not unit_id then
-                qerror("'" .. unit_id_str .. "' is not a valid unit ID.")
-            end
-        elseif arg == valid_args.FILE then
-            i = i + 1 -- consume next arg
-            filename = args[i]
-            if not filename then
-                -- we reached the end of the arguments list
-                qerror("Missing profile name.")
-            end
-        elseif arg == valid_args.PROFILE then
-            i = i + 1 -- consume next arg
-            profile_name = args[i]
-            if not profile_name then
-                -- we reached the end of the arguments list
-                qerror("Missing profile name.")
-            end
-        elseif arg == valid_args.RESET then
-            while args[i + 1] and not contains(valid_args, args[i + 1]) do
-                i = i + 1
-                reset_table[args[i]:upper()] = true
-            end
-        else
-            qerror("'" .. arg .. "' is not a valid argument.")
+    local unit
+    if args.unit then
+        unit = tonumber(args.unit)
+        if not unit then
```

Comment:
You could also use `df.unit.find(unit)` here to catch IDs that are valid numbers but don't refer to valid units (although I suspect the scripts you're calling out to do that already).

Reply:
Yes, the called scripts already do that in their `assign` function.

My reasoning is: the `assign` function in each script can be called from external scripts, so we should check if the arguments passed to it are valid.

The `main` function in each script exists to allow calling the scripts directly via the DFHack console, and so (in my opinion) should just quickly check the correct formatting of the arguments but should not be responsible for the actual validity of the values passed to it, because it's going to call the `assign` function anyway, which will perform the required checks.

Of course, I will modify the code as requested.

Reply:
Makes sense - thanks for the details!